src/components/appNavigation/TabsView.js

```diff
@@ -87,63 +112,51 @@ const TabsView = React.memo((props: TabViewProps) => {
           return newtoken
         })
     }
+
     log.debug('tokens:', { _marketToken, _token })
+
     if (isIOS) {
       setToken(_token)
       setMarketToken(_marketToken)
     }
-  }
+  }, [setToken, setMarketToken])
 
   useEffect(() => {
     if (config.isEToro) {
       fetchTokens()
     }
   }, [])
 
-  const goToRewards = () => {
+  const goToRewards = useCallback(() => {
     if (isIOS) {
       const src = `${config.web3SiteUrl}?token=${token}&purpose=iframe`
       window.open(src, '_blank')
     } else {
       navigation.navigate('Rewards')
     }
-  }
-  const goToSupport = () => {
+  }, [navigation, token])
+
+  const goToSupport = useCallback(() => {
     navigation.navigate('Support')
-  }
-  const goToMarketplace = () => {
+  }, [navigation])
+
+  const goToMarketplace = useCallback(() => {
     if (isIOS) {
       const src = `${config.marketUrl}?jwt=${marketToken}&nofooter=true`
       window.open(src, '_blank')
     } else {
       navigation.navigate('Marketplace')
     }
-  }
+  }, [navigation, marketToken])
 
   return (
     <Appbar.Header dark>
-      {config.isEToro && (
-        <TouchableOpacity testID="rewards_tab" onPress={goToRewards} style={styles.rewardsStyle}>
-          <Icon name="rewards" size={36} color="white" />
-        </TouchableOpacity>
-      )}
-      <Appbar.Content />
-      <TouchableOpacity onPress={goToRewards}>
-        <Icon name="invite2" size={36} color="white" testID="invite_tab" />
-      </TouchableOpacity>
-      <Appbar.Content />
-      {config.market && (
-        <TouchableOpacity testID="goodmarket_tab" onPress={goToMarketplace} style={styles.marketIconBackground}>
-          <Icon name="goodmarket" size={36} color="white" />
-        </TouchableOpacity>
-      )}
-      <Appbar.Content />
-      <TouchableOpacity onPress={goToSupport} style={styles.feedback}>
-        <Icon name="support2" size={36} color="white" testID="support_tab" />
-      </TouchableOpacity>
-      <Appbar.Content />
+      {config.isEToro && etoroButtons(goToRewards, styles)}
```

Comment:
```
const { isEToro, market } = config;

// ...

{isEToro && ...}
```

Reply:
this way ill be shorter & will look better

Reply:
done